Specs/GCM.lean

```diff
@@ -0,0 +1,198 @@
+/-
+Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author(s): Yan Peng
+-/
+import Arm.BitVec
+import Arm.Insts.Common
+
+-- References: https://nvlpubs.nist.gov/nistpubs/legacy/sp/nistspecialpublication800-38d.pdf
+
+namespace GCM
+
+open BitVec
+
+def R : (BitVec 128) := 0b11100001#8 ++ 0b0#120
```

Comment:
Suggestion: Refer to "6.3 Multiplication Operation on Blocks" and comment that this is, up to `x^127`, a representation of the polynomial that the spec uses to define `GF(2^128)`?

Reply:
Done